paddle/platform/enforce.h

```diff
@@ -162,5 +162,11 @@ inline void throw_on_error(T e) {
     }                                                                   \
   } while (0)
 
+#define PADDLE_ENFORCE_EQ(__VAL0, __VAL1, ...)                              \
+  PADDLE_ENFORCE((__VAL0) == (__VAL1),                                      \
```

Comment:
@Superjom like Glog, there is a bug might trigger a compilation warning/error. 

we might need to fix it.

```cpp
template<typename T1, typename T2>
struct min_type {
    static const bool t1_to_t2 = std::is_convertible<T1, T2>::value;
    typedef typename std::conditional<t1_to_t2, T2, T1>::type type;
};

typedef typename min_type<decltype(__VAL0), decltype(__VAL1)>::type T;
PADDLE_ENFORCE(T(__VAL0) == T(__VAL1), ...
```